classes/controller/FrontController.php

```diff
@@ -307,7 +307,7 @@ public function init()
         }
 
         if ($this->auth && !$this->context->customer->isLogged($this->guestAllowed)) {
-            Tools::redirect('index.php?controller=authentication' . ($this->authRedirection ? '&back=' . $this->authRedirection : ''));
+            Tools::redirect('/index.php?controller=authentication' . ($this->authRedirection ? '&back=' . $this->authRedirection : ''));
```

Comment:
What happened if your prestashop is under a subdirectory like `http://website.com/shop`